app/src/main/java/com/example/bookwormadventuresdeluxe2/CreateAccountActivity.java

```diff
@@ -1,3 +1,13 @@
+/**
```

Comment:
Another way to achieve this is to create a method that checks if the email is available. And set the error of username and email editTexts at the same time. 
This would save us a lot of code and from creating and deleting a firebase user.

Reply:
okay i will do a database check of the e-mail like the username!

Reply:
it actually is really hard because we still rely on FBAuth to check if 123 is an invalid email. so it looks like this at the moment and its not fully tested.
![image](https://user-images.githubusercontent.com/39696855/97792127-d7d8ed00-1b9f-11eb-898e-13febeac60b5.png)

Reply:
still working on it

Reply:
i found the solution !

Reply:
nevermind this is rough

Reply:
there is actually no way around this case
![image](https://user-images.githubusercontent.com/39696855/97793654-796a3980-1bb4-11eb-9b77-6f197021a22b.png)

even if i separated the email check it will still break there
sadly i spent 4 hours on it and got nowhere

Reply:
so sorry for the late reply. a little tired from work.
I was thinking of something like this. We could check for email right after checking for username to get around the asynchronous firebase queries. I implemented the new method, take a look and let me know what u think about it.
https://drive.google.com/drive/folders/19ja_4Yu60HmT8sBO2GhQzYuGjsvjDt7n?usp=sharing

Reply:
ok sure ill check it out after i finish this lab

Reply:
Check it out. I ran your code and it runs into this problem:
![image](https://user-images.githubusercontent.com/39696855/97820789-0375d900-1c6d-11eb-8850-8d35861e8dd9.png)

So the only way to get 'invalid email' at the same time as 'username taken' is to run the email through auth no matter what :e

Reply:
thanks for checking.
The case u mentioned is when email editText is not in the right format, "123" here. 
A method to check if the email string pattern is a simple fix. https://stackoverflow.com/questions/6119722/how-to-check-edittexts-text-is-email-address-or-not
let's not spend anymore time with this. 

Current implementation works. Initially, didn't fell good about creating a FB user and deleting when email not available.

Reply:
ill implement this. So we dont run into errors if two people are signing up at the same time (highly unprobably)